packages/salesforcedx-vscode-core/src/commands/functions/functionService.ts

```diff
@@ -180,27 +187,44 @@ export class FunctionService {
    */
   public async debugFunction(rootDir: string) {
     const functionExecution = this.getStartedFunction(rootDir);
-    if (functionExecution) {
-      const { debugPort, debugType } = functionExecution;
-      const debugConfiguration: vscode.DebugConfiguration = {
-        type: debugType,
-        request: 'attach',
-        name: 'Debug Invoke', // This name doesn't surface in UI
-        resolveSourceMapLocations: ['**', '!**/node_modules/**'],
-        console: 'integratedTerminal',
-        internalConsoleOptions: 'openOnSessionStart',
-        localRoot: rootDir,
-        remoteRoot: '/workspace',
-        hostName: '127.0.0.1',
-        port: debugPort
-      };
-      if (!functionExecution.debugSession) {
-        await vscode.debug.startDebugging(
-          getRootWorkspace(),
-          debugConfiguration
-        );
-      }
+    if (!functionExecution) {
+      // TODO: report error?
+      return;
+    }
+
+    if (!functionExecution.debugSession) {
+      const debugConfiguration = this.getDebugConfiguration(functionExecution, rootDir);
+
+      await vscode.debug.startDebugging(
+        getRootWorkspace(),
+        debugConfiguration
+      );
+    }
+  }
+
+  /***
+   * Create a DebugConfiguration object
+   */
+  public getDebugConfiguration(functionExecution: FunctionExecution, rootDir: string): vscode.DebugConfiguration {
+    const { debugPort, debugType } = functionExecution;
+    const debugConfiguration: vscode.DebugConfiguration = {
+      type: debugType,
+      request: 'attach',
+      name: 'Debug Invoke', // This name doesn't surface in UI
+      resolveSourceMapLocations: ['**', '!**/node_modules/**'],
+      console: 'integratedTerminal',
+      internalConsoleOptions: 'openOnSessionStart',
+      localRoot: rootDir,
+      remoteRoot: '/workspace',
+      hostName: '127.0.0.1',
+      port: debugPort
+    };
+
+    if (this.getFunctionType() === functionType.JAVASCRIPT && functionExecution.isContainerless) {
```

Comment:
Interesting! How did you find out this was the change that was required? How is that typescript in containerless mode works even without this fix?

Reply:
Jeff explained this in the elevensies today.